src/todos/reducer.spec.js

```diff
@@ -0,0 +1,166 @@
+import chai, { expect } from 'chai';
+import dirtyChai from 'dirty-chai';
+import { stub, test } from 'sinon';
+import { OrderedMap } from 'immutable';
+import { reducers } from './reducer';
+import { actionTypes as t } from './constants';
+import { actionTypes } from '../generated/todo/';
+
+chai.use(dirtyChai);
+
+describe('Test todo reducer', () => {
+  let todos;
+  let randomStub;
+  let floorStub;
+  beforeEach(() => {
+    todos = reducers.todos;
+    randomStub = stub(Math, 'random', () => 0.5);
+    floorStub = stub(Math, 'floor', () => 1);
+  });
+  afterEach(() => {
+    randomStub.restore();
+    floorStub.restore();
+  });
+
+  describe('ActionType: add', () => {
+    // these are lets as they get reassigned after every test
+    let action;
+    let initState;
+    beforeEach(() => {
+      action = {
+        text: 'some text',
+        type: t.add,
+      };
+      initState = new OrderedMap();
+    });
+
+    it('should add an item to the map', test(() => {
+      expect(todos(initState, action).size).to.equal(initState.size + 1);
+    }));
+    it('should have a uuid associated with it', test(() => {
+      expect(todos(initState, action)).to.not.be.undefined();
+    }));
+    it('should add a todo with the correct text', () => {
+      expect(todos(initState, action).get(1).text).to.equal('some text');
+    });
+    it('should add a todo with the completed property set to false', () => {
+      expect(todos(initState, action).get(1).completed).to.equal(false);
+    });
+  });
+  describe('ActionType: delete', () => {
+    it('should remove the todo from the list of todos', () => {
+      const action = {
+        type: t.delete,
+        id: 1,
```

Comment:
Should change this to `payload`, to conform to [FSA](https://github.com/acdlite/flux-standard-action)